src/compute/src/render/join/mz_join_core.rs

```diff
@@ -477,7 +488,7 @@ where
         }
 
         if !temp.is_empty() {
-            *fuel += flush(temp, &mut session);
+            *work -= flush(temp, &mut session);
```

Comment:
Curious: what happens if flush would not refill fuel? This might solve the issue when data consolidates really well, but I am on mobile so it's hard to see!

Reply:
It would solve the issue of the join not yielding when the output data consolidates well. However, the same issue also exists when the `result` closure filters aggressively, and that would still be a problem.

We could fix both by counting the number of inputs to `result`, rather than the number of outputs. There are two reasons I didn't consider that here:

* It would be inconsistent with how the DD join operators behave. Both count output records because in DD yielding is only intended to prevent OOMs, not to ensure interactivity.
* I didn't want to change the existing (mostly working) behavior by default, so we can take our time to test the new yielding behavior.

I think time-based yielding is better than work-based for us. Even if we adjusted the work-based strategy, we would still run into instances where the `result` closure is particularly expensive and we yield less often than we'd like.